Source/OCMock/OCMRealObjectForwarder.m

```diff
@@ -42,6 +45,33 @@ - (void)handleInvocation:(NSInvocation *)anInvocation
 	}
 
 	[anInvocation invoke];
+	if(isInInitFamily || isInCreateFamily)
+	{
+	    // OCMInvocationStub does some post processing after all of the invocation actions are called
+	    // to make sure that init family and create family calls are handled correctly with regards
+	    // to retain/release. In the case where we are forwarded to a real object, the handling
+	    // that the real object has already done needs to be "undone" so we don't over retain or under
+	    // release.
+	    id returnVal;
+	    [anInvocation getReturnValue:&returnVal];
+	    if (isInCreateFamily)
+	    {
+	        // methods that "create" an object will return it with an extra retain count
+	        // autorelease it so that when OCMInvocationStub retains it that we balance.
+	        [returnVal autorelease];
+	    }
+	    else if(isInInitFamily)
+	    {
+	        // init family methods "consume" self and retain their return value.
+	        // Retain the target (that OCMInvocationStub will release) and autorelease the returnVal
```

Comment:
This comment is misleading. In this method (line 38) we're setting the target of the invocation to the real object. This is what we are retaining here. However, the stub copied the original target, which was the mock, and releases that. So, the retain here does not balance the release in the stub. I don't know why this still works. Noticed it when refactoring the code.